### PR TITLE
Static Execution Adapter

### PR DESCRIPTION
**Pull request by peterschwarz**
First pass implementation of the static execution adapter and all of the necessary changes to make it work.

Replaces #32, #47

### CODE REVIEW

**Comment by peterschwarz** [comment]
Now dependent on #50

File: libtransact/src/execution/adapter/static_adapter.rs [hunk 1]
```diff
@@ -0,0 +1,628 @@
...
+
+    impl CommandTransactionHandler {
+        pub fn new() -> Self {
+            CommandTransactionHandler {
+                versions: vec![COMMAND_VERSION.to_owned()],
+            }
+        }
+    }
+
+    impl TransactionHandler<String, Vec<u8>> for CommandTransactionHandler {
```
Comment (vaporos) [inline libtransact/src/execution/adapter/static_adapter.rs]: Should we create a top-level handlers module? We could enable this for tests and/or a feature.

Comment (peterschwarz) [reply libtransact/src/execution/adapter/static_adapter.rs]: Sounds like something for a future PR.

File: libtransact/src/execution/adapter/static_adapter.rs [hunk 2]
```diff
@@ -0,0 +1,628 @@
...
+                    transaction_id: txn_id,
+                    error_message: "Test Fail Succeeded".into(),
+                    error_data: vec![],
+                }
+            ),
+            result.unwrap()
+        );
+    }
+
+    fn make_transaction(commands: &[Command]) -> TransactionPair {
```
[the remark this answers is not in the document]
Comment (peterschwarz) [reply libtransact/src/execution/adapter/static_adapter.rs]: Sounds like something for a future PR

File: libtransact/src/execution/adapter/static_adapter.rs [hunk 3]
```diff
@@ -0,0 +1,628 @@
...
+                    .any(|v| v == family.family_version())
+        });
+
+        if handler.is_none() {
+            on_done(Err(ExecutionAdapterError::RoutingError(Box::new(
+                transaction_pair,
+            ))));
+        } else {
+            let context = StaticTransactionContext::new(self.context_manager.clone(), context_id);
+            match handler.unwrap().apply(&transaction_pair, &context) {
```
[the remark this answers is not in the document]
Comment (peterschwarz) [reply libtransact/src/execution/adapter/static_adapter.rs]: Yes, because we checked that the value was none in the if clause.

**Comment by peterschwarz** [comment]
This will be broken up into multiple PR's (not unlike #57).

**Pull request closed without merging**